code/iaas/logic/src/main/java/io/cattle/platform/process/instance/InstancePreCreate.java

```diff
@@ -34,12 +36,15 @@
 import org.apache.commons.lang3.StringUtils;
 
 import com.google.common.base.Joiner;
+import com.netflix.config.DynamicStringProperty;
 
 @Named
 public class InstancePreCreate extends AbstractObjectProcessLogic implements ProcessPreListener, Priority {
     @Inject
     JsonMapper jsonMapper;
 
+    public static final DynamicStringProperty DEFAULT_REGISTRY = ArchaiusUtil.getString("default.registry");
```

Comment:
I did the above changes. I had included the code to prepend default registry earlier, but I don't know how I missed it in the PR, I'm including that code in the new changes